src/stacking.rs

```diff
@@ -0,0 +1,52 @@
+
+use imp_prelude::*;
+use error::{ShapeError, ErrorKind, from_kind};
+
+/// Stack arrays along the given axis.
+///
+/// *Panics* if axis is out of bounds.
+pub fn stack<'a, A, D>(arrays: &[ArrayView<'a, A, D>], axis: Axis)
+    -> Result<OwnedArray<A, D>, ShapeError>
+    where A: Copy,
+          D: Dimension + RemoveAxis
+{
+    if arrays.len() == 0 {
+        return Err(from_kind(ErrorKind::Unsupported));
+    }
+    let mut res_dim = arrays[0].dim().clone();
+    let common_dim = res_dim.remove_axis(axis);
+    if arrays.iter().any(|a| a.dim().remove_axis(axis) != common_dim) {
+        return Err(from_kind(ErrorKind::IncompatibleShape));
+    }
+
+    let stacked_dim = arrays.iter()
+                            .fold(0, |acc, a| acc + a.dim().index(axis));
+    *res_dim.index_mut(axis) = stacked_dim;
+
+    // we can safely use uninitialized values here because they are Copy
+    // and we will only ever write to them
+    let size = res_dim.size();
+    let mut v = Vec::with_capacity(size);
+    unsafe {
+        v.set_len(size);
+    }
+    let mut res = try!(OwnedArray::from_vec_dim(res_dim, v));
+
+    {
+        let mut assign_view = res.view_mut();
+        for array in arrays {
+            let len = *array.dim().index(axis);
+            let (mut front, rest) = assign_view.split_at(axis, len);
+            front.assign(array);
+            assign_view = rest;
+        }
+    }
+    Ok(res)
+}
+
+#[macro_export]
+macro_rules! stack {
+    ([ $( $a:expr ),* ]; $axis:expr) => {
+        ndarray::stack(&[ $($a.view() ),* ], $axis).unwrap()
+    }
```

Comment:
I think using the new `ArrayView::from` works well. Shouldn't do much different than the .view() call. But .view() works well too? View accepts all kinds of arrays and references to arrays.

Should we make the syntax lighter? stack![Axis(0), a, b, c] would work too I think. And using `+` to enforce that there is at least one array.